utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/swagger/SwaggerTyped.scala

```diff
@@ -188,6 +196,20 @@ object SwaggerTyped {
       TypedObjectTypingResult(elementType.mapValuesNow(typingResult).toList.sortBy { case (propertyName, _) => propertyName })
     }
   }
+
+  private def inferredIntType(minValue: BigDecimal, exclusiveMinValue: BigDecimal, maxValue: BigDecimal, exclusiveMaxValue: BigDecimal): SwaggerTyped = {
+
+    val lowerBoundary: Option[BigDecimal] = List(Option(exclusiveMinValue).map(_ + 1), Option(minValue)).flatten.sorted(Ordering.BigDecimal.reverse).headOption
+    val upperBoundary: Option[BigDecimal] = List(Option(exclusiveMaxValue).map(_ - 1), Option(maxValue)).flatten.sorted(Ordering.BigDecimal).headOption
+
+    List(lowerBoundary, upperBoundary).flatten match {
+      case min :: max :: Nil if min.isValidInt && max.isValidInt => SwaggerInteger
+      case min :: max :: Nil if min.isValidLong && max.isValidLong => SwaggerLong
```

Comment:
1. What about situations:
a) one is validInt and the second isValidLong?
b) both are the same validInt
2. Maybe it would be better to replace those min, max matching by `list.forAll(_.isValidInt)` etc. Why min/max order is important? What about cases where min is a larger negative number?

Reply:
After rethinking it is ok :)